src/components/calcite-inline-editable/calcite-inline-editable.tsx

```diff
@@ -0,0 +1,265 @@
+    if (!this.editingEnabled) this.htmlInput.tabIndex = -1;
+  }
+
+  render(): VNode {
+    return (
+      <Host>
+        <div
+          class="calcite-inline-editable-wrapper"
+          onClick={this.enableEditingHandler}
+          onKeyDown={this.escapeKeyHandler}
+        >
+          <div class="calcite-inline-editable-input-wrapper">
+            <slot />
+          </div>
+          <div class="calcite-inline-editable-controls-wrapper">
+            {!this.editingEnabled && (
+              <calcite-button
+                appearance="transparent"
+                aria-label={this.intlEnableEditing}
+                class="calcite-inline-editable-enable-editing-button"
+                color="dark"
+                disabled={this.inputElement.disabled}
+                iconStart="pencil"
+                onClick={this.enableEditingHandler}
+                ref={(el) => (this.enableEditingButton = el)}
+                scale={this.scale}
+                theme={this.theme}
+              />
+            )}
+            {this.shouldShowControls && [
+              <div class="calcite-inline-editable-cancel-editing-button-wrapper">
+                <calcite-button
+                  appearance="transparent"
+                  aria-label={this.intlCancelEditing}
+                  class="calcite-inline-editable-cancel-editing-button"
+                  color="dark"
+                  disabled={this.inputElement.disabled}
+                  iconStart="x"
+                  onClick={this.cancelEditingHandler}
+                  scale={this.scale}
+                  theme={this.theme}
+                />
+              </div>,
+              <calcite-button
+                appearance="solid"
+                aria-label={this.intlConfirmChanges}
+                class="calcite-inline-editable-confirm-changes-button"
+                color="blue"
+                disabled={this.inputElement.disabled}
+                iconStart="check"
+                loading={this.loading}
+                onClick={this.confirmChangesChangesHandler}
+                scale={this.scale}
+                theme={this.theme}
+              />
+            ]}
+          </div>
+        </div>
+      </Host>
+    );
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  @Event() calciteInlineEditableCancelEditing: EventEmitter;
+
+  @Event() calciteInlineEditableConfirmChanges: EventEmitter;
+
+  @Event() calciteInlineEditableEnableEditing: EventEmitter;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("calciteInputBlur")
+  blurHandler(): void {
+    if (!this.hasControls) this.disableEditing();
+  }
+
+  @Listen("click", { target: "window" })
```

Comment:
maybe this logic could be moved to the label instead of the input?

Reply:
I'm going to punt on this for now, but if this presents problems we can definitely revisit.